### PR TITLE
[stable/prometheus-blackbox-exporter] Breaking change: general upgrade and ServiceMonitor

### PR DESCRIPTION
**Pull request by desaintmartin**
- Update deployment to apps/v1.
- Breaking change: use recommended labels.
- Add optional ServiceMonitor.
- Update image to 0.13.0.
- Add desaintmartin to owners.

I can break down those changes into several PRs if you want.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/prometheus-blackbox-exporter/README.md [hunk 1]
```diff
@@ -70,6 +70,11 @@ The following table lists the configurable parameters of the Blackbox-Exporter c
 | `service.type`                         | type of service to create                       | `ClusterIP`                   |
 | `service.port`                         | port for the blackbox http service              | `9115`                        |
 | `service.externalIPs`                  | list of external ips                            | []                            |
+| `serviceMonitor.enabled`               | If true, a ServiceMonitor CRD is created for a prometheus operator | `false`    |
+| `serviceMonitor.labels`                | Labels for prometheus operator                  | `{}`                          |
+| `serviceMonitor.interval`              | Interval for prometheus operator endpoint       | `30s`                        |
+| `serviceMonitor.url`                   | The URL that blackbox will scrape if serviceMonitor is enabled | `http://www.changemeoriwillfail.com/healthz` |
```
Comment (desaintmartin) [inline stable/prometheus-blackbox-exporter/README.md]: Tell me if you see something that is more elegant but still validates with `helm lint`.

File: stable/prometheus-blackbox-exporter/templates/servicemonitor.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    port: http
+    params:
+      module:
+      - http_2xx
+      target:
+      - {{ .Values.serviceMonitor.url }}
+    path: "/probe"
+    targetPort: 9115
+    metricRelabelings:
+      - sourceLabels: [__address__]
```
Comment (desaintmartin) [inline stable/prometheus-blackbox-exporter/templates/servicemonitor.yaml:27]: I'm open to suggestions for this, as it is quite opiniated.

**Comment by desaintmartin** [comment]
up?

**Comment by mattfarina** [comment]
@desaintmartin The DCO isn't on the old commits. If you go to the checks tab there is a command you can run which will help you sign all of them.

**Comment by desaintmartin** [comment]
Oops, done, thanks.

File: stable/prometheus-blackbox-exporter/README.md (hunk 1 above)
Comment (mattfarina) [reply stable/prometheus-blackbox-exporter/README.md]: Oh, this should be `example.com` instead. Saw a funny tweet where someone bought a DNS name and broke things for people.... https://twitter.com/JGamblin/status/1100414626259324929

example.com is a reserved domain. It wont be bought or hijacked. See https://www.iana.org/domains/reserved

**Comment by desaintmartin** [comment]
Up?

**Comment by mattfarina** [comment]
cc @gianrubio

@desaintmartin Can you rebase from master

**Comment by desaintmartin** [comment]
Rebased, removed useless commit to upgrade to 0.13.0.

**Comment by desaintmartin** [comment]
Not stale

**Comment by desaintmartin** [comment]
Anybody to review, even to refuse?

**Comment by cablespaghetti** [comment]
Good work! This PR is a must for me; I manage everything Prometheus-related with the Prometheus Operator. However the limitation of only being able to have a single URL for the ServiceMonitor isn't great. I'll report back when I get multiple URLs working.

**Comment by cablespaghetti** [comment]
This is what I changed to get multiple URLs working. Sorry I can't work out how to get pretty Github diffs to work.

```
diff --git a/stable/prometheus-blackbox-exporter/README.md b/stable/prometheus-blackbox-exporter/README.md
index 0a8b03b5a..43ed35c20 100644
--- a/stable/prometheus-blackbox-exporter/README.md
+++ b/stable/prometheus-blackbox-exporter/README.md
@@ -75,8 +75,7 @@ The following table lists the configurable parameters of the Blackbox-Exporter c
 | `serviceMonitor.enabled`               | If true, a ServiceMonitor CRD is created for a prometheus operator | `false`      |
 | `serviceMonitor.labels`                | Labels for prometheus operator                    | `{}`                          |
 | `serviceMonitor.interval`              | Interval for prometheus operator endpoint         | `30s`                         |
-| `serviceMonitor.url`                   | The URL that blackbox will scrape if serviceMonitor is enabled | `http://example.com/healthz` |
-| `serviceMonitor.urlHumanReadable`      | Optional human readable URL that will appear in Prometheus / AlertManager | `nil` |
+| `serviceMonitor.urls`                  | The list of URLs that blackbox will scrape if serviceMonitor is enabled | []      |
 
 Specify each parameter using the `--set key=value[,key=value]` argument to `helm install`. For example,
 
diff --git a/stable/prometheus-blackbox-exporter/templates/servicemonitor.yaml b/stable/prometheus-blackbox-exporter/templates/servicemonitor.yaml
index 84b52f842..9522b5309 100644
--- a/stable/prometheus-blackbox-exporter/templates/servicemonitor.yaml
+++ b/stable/prometheus-blackbox-exporter/templates/servicemonitor.yaml
@@ -13,27 +13,31 @@ metadata:
     {{- end }}
 spec:
   endpoints:
+  {{- $interval := .Values.serviceMonitor.interval }}
+  {{- $scrapeTimeout := .Values.serviceMonitor.scrapeTimeout }}
+  {{- range .Values.serviceMonitor.urls }}
   - port: http
     scheme: http
     path: "/probe"
-    interval: {{ .Values.serviceMonitor.interval }}
-    scrapeTimeout: {{ .Values.serviceMonitor.scrapeTimeout }}
+    interval: {{ $interval }}
+    scrapeTimeout: {{ $scrapeTimeout }}
     params:
       module:
       - http_2xx
       target:
-      - {{ .Values.serviceMonitor.url }}
+      - {{ .url}}
     metricRelabelings:
       - sourceLabels: [__address__]
         targetLabel: __param_target
       - sourceLabels: [__param_target]
         targetLabel: instance
       - targetLabel: target
-        {{- if .Values.serviceMonitor.urlHumanReadable }}
-        replacement: {{ .Values.serviceMonitor.urlHumanReadable }}
+        {{- if .urlHumanReadable }}
+        replacement: {{ .urlHumanReadable }}
         {{- else }}
-        replacement: {{ .Values.serviceMonitor.url }}
+        replacement: {{ .url }}
         {{- end }}
+  {{- end }}
   jobLabel: "{{ .Release.Name }}"
   selector:
     matchLabels:
diff --git a/stable/prometheus-blackbox-exporter/values.yaml b/stable/prometheus-blackbox-exporter/values.yaml
index d3b96bec7..deb7fbfea 100644
--- a/stable/prometheus-blackbox-exporter/values.yaml
+++ b/stable/prometheus-blackbox-exporter/values.yaml
@@ -89,7 +89,8 @@ serviceMonitor:
   labels: {}
   interval: 30s
   scrapeTimeout: 30s
-  # The URL that blackbox will scrape
-  url: http://example.com/healthz
-  # Optional human readable URL that will appear in Prometheus / AlertManager
-  urlHumanReadable:  # www.changemeoriwillfail.com
+  urls:
+    # The URL that blackbox will scrape
+    # - url: http://example.com/healthz
+    # Optional human readable URL that will appear in Prometheus / AlertManager
+    # urlHumanReadable:  # www.changemeoriwillfail.com
```

**Review by gianrubio — APPROVED** [review]
/ok-to-test
/lgtm

**Comment by desaintmartin** [comment]
@cablespaghetti feel free to open a PR to add support for multiple URLs!

**Comment by cablespaghetti** [comment]
That's great. Glad this got merged. I'm going to add support for PrometheusRules as well so the alerting can be all deployed at once.

**Pull request merged**